app/models/vim_performance_analysis.rb

```diff
@@ -439,6 +439,107 @@ def self.find_perf_for_time_period(obj, interval_name, options = {})
                   .select(options[:select])
   end
 
+  def self.collect_metrics(obj, start_time, end_time, interval)
+    client = ManageIQ::Providers::Kubernetes::ContainerManager::MetricsCapture::CaptureContext.new(
```

Comment:
@simon3z @chessbyte yes, I will move this to the provider.
I also noticed this problem, when I started to write a spec and it didn't make sense ...